packages/garbo/src/tasks/cockroachPrep.ts

```diff
@@ -16,35 +15,84 @@ import {
 import {
   abort,
   adv1,
+  Effect,
+  effectModifier,
+  isShruggable,
+  Item,
   mallPrice,
   myBuffedstat,
+  myEffects,
+  retrieveItem,
   runChoice,
+  Stat,
+  toEffect,
+  use,
   visitUrl,
 } from "kolmafia";
 import { garboValue } from "../garboValue";
 import { freeFightFamiliar } from "../familiar";
-import { freeFightOutfit } from "../outfit";
+import { freeFightOutfit, meatTargetOutfit } from "../outfit";
 import { GarboStrategy, Macro } from "../combat";
 import { acquire } from "../acquire";
 
+function pickBestDebuff(stat: Stat): Item {
+  const statName = stat.toString();
+
+  const debuffMenu = Item.all()
+    .filter((it) => {
+      const effect = effectModifier(it, "Effect").toString();
+      return effect && !(effect in myEffects());
+    })
+    .filter((it) => getModifier(statName, it) < 0)
+    .map(
+      (it) => [it, mallPrice(it) / getModifier(statName, it)] as [Item, number],
+    ); // getModifier should return a negative value, flipping these negative
+
+  // ...so that when we maxBy we pick the smallest negative value, spending the least meat
+  return maxBy(debuffMenu, ([, value]) => value)[0];
+}
+
 // Just checking for the gummi effects for now, maybe can check other stuff later?
 function checkAndFixOvercapStats(): void {
-  if (myBuffedstat($stat`Muscle`) >= 100) {
-    if (have($effect`Gummiheart`)) uneffect($effect`Gummiheart`);
-  }
-  if (myBuffedstat($stat`Mysticality`) >= 100) {
-    if (have($effect`Gummibrain`)) uneffect($effect`Gummibrain`);
-  }
-  if (myBuffedstat($stat`Moxie`) >= 100) {
-    if (have($effect`Gummiskin`)) uneffect($effect`Gummiskin`);
-  }
-  if (
-    myBuffedstat($stat`Moxie`) >= 100 ||
-    myBuffedstat($stat`Mysticality`) >= 100 ||
-    myBuffedstat($stat`Muscle`) >= 100
-  ) {
-    uneffect($effect`Having a Ball!`);
+  const stats = Stat.all();
+  const effects: Effect[] = Object.keys(myEffects()).map((effectName) =>
+    toEffect(effectName),
+  );
+
+  // Use a traditional for loop for stats
+  for (let i = 0; i < stats.length; i++) {
+    const stat = stats[i];
+    const statName = stat.toString();
+
+    while (myBuffedstat(stat) > 100) {
+      for (const isShruggablePass of [true, false]) {
+        for (let j = 0; j < effects.length; j++) {
```

Comment:
We absolutely don't need to use a nested loop here, we can achieve this much more efficiently by just handling things consecutively